### ISSUE TITLE
Incorrect reporting of target size violation from incorrect offset calculation

### ISSUE THREAD

**Issue by RyanBerliner** [dequelabs/axe-core#4295]
### Product

axe-core

### Product Version

4.8.3

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation

I would expect no target size violations to occur in the example.

#### Actual

The link labelled "Incorrectly flagged as target size violation" is in violation of target size, claiming it has 0 offset from both of the the other links.

![IMG_2406](https://github.com/dequelabs/axe-core/assets/22206986/34b05eb2-2a6b-49f1-a24d-d1e7c44a5a78)

#### How to Reproduce

Unfortunately I was not able to reproduce this in jsfiddle (or the like), so the following html will need to be saved and opened as a file in your browser. All instructions to reproduce are in the comment in the code snippet.

```html
<!DOCTYPE html>
<html lang="en-US">
	<head>
		<title>Target Size</title>
	</head>
​
	<body>
		<div style="max-width:290px">
			<p>
				<a href="#">lorem ipsum dolor</a>
				<a href="#">amet</a>
				<span>et dolore magna aliqua</span>
			</p>
			<a href="#" style="display: inline-block;">Incorrectly flagged as target size violation</a>
		</div>
	</body>
​
	<!--
​
	In your browser, do the following (could not replicate in jsfiddle or the like)
​
	1. Open this HTML file
	2. Open the devtools console on that page - the rest of the instruction will take place there.
	3. Paste in axe source to make the global `axe` available on the page
​
	   For example: https://www.unpkg.com/axe-core@4.8.3/axe.min.js
​
	4. Configure axe to ONLY report on target size violations
​
		axe.configure({
			rules: [{id: 'target-size', enabled: true}],
			disableOtherRules: true
		})
​
	5. Run axe and print out the violations
​
		axe.run().then(r => console.log(r.violations));
​
	At this point you will observe that the last link is flagged as being in violation
	of target size. Specifically, it states that it has 0 offset from the other 2 links
	on the page, despite clearly being far away from them.
​
	This can be replicated in Chrome, Firefox, Safari
​
	-->
</html>
```

#### Additional context

The test case above is very carefully constructed. Here is what I found to be important about it specifically (and what is not)

- the div must be sized such that the paragraph wraps to multiple lines
- re: the first paragraph
  - The 2 links in here must be close enough to each other that they _should_ be flagged as violating offset limitations _if_ the were inline-block (its my understanding inline elements are not caught by axe present day for target-size violations?)
  - The text after the links _must_ be wrapped in a span
  - This span _must_ break in the middle as part of the wrapping forced by the div
- The final link outside the paragraph  _must_ be `display: inline-block;`

**Comment by dbjorge** [comment]
See related https://github.com/dequelabs/axe-core/issues/4194 (same root cause manifesting as one part of a different issue)

**Comment by RyanBerliner** [comment]
Please let me know if you'd appreciate me dedicating some time to work on a PR here, happy to help where I can.

**Comment by nicksergeant** [comment]
I think we're running into this as well:

![Image](https://github.com/user-attachments/assets/e81cf19e-18b6-4506-9eb9-2f6df5eab8ae)

I've noticed this is somewhat flaky - sometimes this violation doesn't trigger but often it does. You can reproduce by visiting https://trumansburg-ny.gov/departments/zoning-and-code-enforcement/ and running:

```
fetch('https://cdnjs.cloudflare.com/ajax/libs/axe-core/4.8.2/axe.min.js').then(r => r.text()).then(t => eval(t)).then(() => axe.run(document, {runOnly: ['wcag2a', 'wcag2aa', 'wcag2aa', 'wcag21a', 'wcag21aa', 'wcag22aa'], iframes: true})).then(results => console.log(results.violations))
```

```
[
    {
        "id": "target-size",
        "impact": "serious",
        "tags": [
            "cat.sensory-and-visual-cues",
            "wcag22aa",
            "wcag258"
        ],
        "description": "Ensure touch target have sufficient size and space",
        "help": "All touch targets must be 24px large, or leave sufficient space",
        "helpUrl": "https://dequeuniversity.com/rules/axe/4.8/target-size?application=axeAPI",
        "nodes": [
            {
                "any": [
                    {
                        "id": "target-size",
                        "data": {
                            "minSize": 24,
                            "width": 316.1,
                            "height": 20
                        },
                        "relatedNodes": [],
                        "impact": "serious",
                        "message": "Target has insufficient size (316.1px by 20px, should be at least 24px by 24px)"
                    },
                    {
                        "id": "target-offset",
                        "data": {
                            "closestOffset": 0,
                            "minOffset": 24
                        },
                        "relatedNodes": [
                            {
                                "html": "<a href=\"/\">Home</a>",
                                "target": [
                                    ".second-menu > li:nth-child(1) > a[href=\"/\"]"
                                ]
                            }
                        ],
                        "impact": "serious",
                        "message": "Target has insufficient space to its closest neighbors. Safe clickable space has a diameter of 0px instead of at least 24px."
                    }
                ],
                "all": [],
                "none": [],
                "impact": "serious",
                "html": "<a href=\"mailto:codeenforcement@trumansburg-ny.gov\">codeenforcement@trumansburg-ny.gov</a>",
                "target": [
                    "p[data-block-key=\"7gzps\"] > a"
                ],
                "failureSummary": "Fix any of the following:\n  Target has insufficient size (316.1px by 20px, should be at least 24px by 24px)\n  Target has insufficient space to its closest neighbors. Safe clickable space has a diameter of 0px instead of at least 24px."
            }
        ]
    }
]
```
